### DISCUSSION TITLE
MapboxMap symbols or circles not displaying (Flutter)

### DISCUSSION THREAD

**Discussion by spiteyyPointyy** [community/community#12109] [Other Feature Feedback, Questions, & Ideas] [answered]
I am trying to display markers on a map after a user clicks a button. I've tries using either the controller.addSymbol or controller.addCircle methods for MapBox_gl however after clicking the button no markers are displayed. I have attached screenshots of the relevant code and errors I received. Any help would be appreciated!

<img width="470" alt="Screenshot 2022-02-25 at 19 18 53" src="https://user-images.githubusercontent.com/100440879/155775057-85397d38-0421-4fce-8667-6f978a33acf7.png">
<img width="239" alt="Screenshot 2022-02-25 at 19 19 17" src="https://user-images.githubusercontent.com/100440879/155775079-d3b1049c-3c53-4722-8918-acf5373f256f.png">
<img width="548" alt="Screenshot 2022-02-25 at 19 19 01" src="https://user-images.githubusercontent.com/100440879/155775094-ede7be4f-758a-4faf-9cf2-d205b428acea.png">
<img width="958" alt="Screenshot 2022-02-25 at 19 14 35" src="https://user-images.githubusercontent.com/100440879/155775162-41ad9dd9-bce5-4892-8da2-a6391c19eabc.png">
<img width="889" alt="Screenshot 2022-02-25 at 19 15 01" src="https://user-images.githubusercontent.com/100440879/155775720-dddfdc39-e7c9-4524-91c7-77f38be09456.png">
<img width="945" alt="Screenshot 2022-02-25 at 19 27 34" src="https://user-images.githubusercontent.com/100440879/155777263-46282e39-45c0-4424-b497-4af306c40b55.png">

**Comment by Imperial-lord** [comment, marked as answer]
Hey @spiteyyPointyy. The addSymbol is an asynchronous task. For instance look at the following code - 

```
_onStyleLoadedCallback() async {
    for (CameraPosition _kRestaurant in _kRestaurantsList) {
      await controller.addSymbol(
        SymbolOptions(
          geometry: _kRestaurant.target,
          iconSize: 0.2,
          iconImage: "assets/icon/food.png",
        ),
      );
    }
    _addSourceAndLineLayer(0, false);
  }
```

Your place markers function needs to be asynchronous, and use the await keyword, before addSymbol() method. 
Secondly, you need to make sure that the floating action button is not pressed before the map styles are loaded, so you should have a flag, which you can set inside the `onStyleLoadedCallback()` and then accordingly show your floating action button. Here is what I mean - 

```
class _MyMapState extends State<MyMap>{
  // 
  // bunch of lines ...
  //
      
  bool showFab = false;
  
  _onStyleLoadedCallback() async{
    setState(() => showFab = true);
  }
  
  // Widget build 
  // and other UI 
  // and finally
          child: (showFab)? FloatingActionButton():Container()

}

I hope that helps thanks for reaching out. Btw, you could have opened a thread here https://github.com/Imperial-lord/mapbox-flutter/discussions instead. Let me know if there is any further issue. 😇

**Reply by spiteyyPointyy** [reply]
Thank you so much for your help, I really appreciate it!

I've made the changes that you had suggested, and switched to using Symbols rather than Circles. It seems that my code does create the symbols now however does not display them on the map. Wondering if I missed something out or if my icons are not being loaded... :/
I shall attach my new code snippets. 

mapscreen.dart: 
<img width="508" alt="Screenshot 2022-02-27 at 13 15 31" src="https://user-images.githubusercontent.com/100440879/155884933-57eb3b88-af4a-4b2b-91ac-5f9158b48acc.png">

<img width="661" alt="Screenshot 2022-02-27 at 13 34 17" src="https://user-images.githubusercontent.com/100440879/155885019-b27d3e1e-02ac-4bd6-8689-33c087f85081.png">


Pubspec.yaml:
<img width="492" alt="Screenshot 2022-02-27 at 13 16 35" src="https://user-images.githubusercontent.com/100440879/155884944-35253d9e-459f-44fa-80b6-d5e1b162fa67.png">
 
Assets/icon/red.png: 
<img width="688" alt="Screenshot 2022-02-27 at 13 46 24" src="https://user-images.githubusercontent.com/100440879/155884985-691e2f81-4723-4a0a-ba2f-e05b6551555b.png">

Thank you so much for helping me once again..

**Reply by spiteyyPointyy** [reply]
No worries, I've managed to fix it... I believe it was showing but there was just some issues with zoom. Thank you so much for your help!

**Reply by Imperial-lord** [reply]
Oh! That is great to know! 😇
